osu.Game/Overlays/BeatmapSet/BeatmapSetHeaderContent.cs

```diff
@@ -222,13 +228,13 @@ private void load(OverlayColourProvider colourProvider)
         {
             coverGradient.Colour = ColourInfo.GradientVertical(colourProvider.Background6.Opacity(0.3f), colourProvider.Background6.Opacity(0.8f));
 
-            State.BindValueChanged(_ => updateDownloadButtons());
-
             BeatmapSet.BindValueChanged(setInfo =>
             {
                 Picker.BeatmapSet = rulesetSelector.BeatmapSet = author.BeatmapSet = beatmapAvailability.BeatmapSet = Details.BeatmapSet = setInfo.NewValue;
                 cover.OnlineInfo = setInfo.NewValue?.OnlineInfo;
 
+                downloadTracker?.RemoveAndDisposeImmediately();
```

Comment:
Is it even added to the hierarchy in this class at all? Doesn't look to be from what I'm reading.

Reply:
Nice catch, this is an oversight.